view/frontend/templates/subscription_freq_selector_product_page.phtml

```diff
@@ -0,0 +1,183 @@
+                            return;
+                        }
+                    }
+
+                    options[name] = value;
+                });
+                options['storeId'] = '<?= /* @noEscape */ $block->getStoreId(); ?>';
+                var formKey = $.mage.cookies.get('form_key');
+                if (!formKey && $('[name="form_key"]').length) {
+                    formKey = $('[name="form_key"]').val();
+                }
+                options['form_key'] = formKey;
+                var items = [];
+                <?php if ($block->isGrouped()): ?>
+                    <?php foreach ($block->getGroupedProductChildren() as $childProduct):?>
+                        var childProductQty = getGroupedProductChildQty(<?= /* @noEscape */ $childProduct->getId(); ?>);
+                        if (childProductQty > 0) {
+                            <?php if ($childProduct->isVirtual()): ?>
+                                var productType = 'digital';
+                            <?php else: ?>
+                                var productType = 'physical';
+                            <?php endif; ?>
+                            items.push({
+                                reference: '<?= /* @noEscape */ $childProduct->getId(); ?>',
+                                price: '<?= /* @noEscape */ $childProduct->getPrice() ?>',
+                                name: '<?= /* @noEscape */ addcslashes($childProduct->getName(), "'"); ?>',
+                                quantity: childProductQty,
+                                options: JSON.stringify(options),
+                                type: productType
+                            });
+                        }
+                    <?php endforeach; ?>
+                <?php else: ?>
+                    <?php if ($block->getProduct()->isVirtual()): ?>
+                        var productType = 'digital';
+                    <?php else: ?>
+                        var productType = 'physical';
+                    <?php endif; ?>
+                    items.push({
+                        reference: '<?= /* @noEscape */ $block->getProduct()->getId(); ?>',
+                        price: <?= /* @noEscape */ $block->getProduct()->getFinalPrice(); ?>,
+                        name: '<?= /* @noEscape */ addcslashes($block->getProduct()->getName(), "'"); ?>',
+                        quantity: getQty(),
+                        options: JSON.stringify(options),
+                        type: productType
+                    });
+                <?php endif; ?>
+
+                return items;
+        };
+
+        var cart;
+        var setupProductPage = function () {
+            cart = {
+                items: getItemsData()
+            };
+
+            // if connect.js is not loaded postpone until it is
+            whenDefined(window, 'BoltCheckout', function(){
+                BoltCheckout.configureProductFrequencySelector(cart, null, callbacks);
```

Comment:
We will be sending a a PR to create `configureProductFrequencySelector()` on Source/storm side very soon!

Reply:
@Nitsorn Is it ok to merge the changes in storm first and then this one?
We are not to release during code freeze so we have some time.

Reply:
yes definitely, we are in the process of getting those in. We certainly have time, but want you to take a look, even if we dont merge, to know that we are on the right path.

Reply:
Sure. Our team will check during next few days

Reply:
please and thank you!